services/storage/src/simcore_service_storage/handlers_files.py

```diff
@@ -287,10 +290,11 @@ async def complete_upload_file(request: web.Request) -> web.Response:
         )
         .with_query(user_id=query_params.user_id)
     )
+    complete_task_state_url = AnyUrl(
```

Comment:
IMO this is not how  `AnyUrl` is suppose to be used. It is a pydantic constraint type designed to annotate model's fields. If you want to use it outside of that context, I suggest you that you use `AnyUrl.build` class method instead, which is what they use internaly to run the validation 

This is why the constructor is "redundant" and you had to add the scheme twice!

```python
AnyUrl.build(scheme=request.url.scheme, host=ip_addr, port=port, path=route_ppath, query=route_query)

Or even better ... I notice that `route` is itself a convenient object to build paths

It would be something like (please double check details in [manual](https://yarl.aio-libs.org/en/latest/api/#yarl-api))
```python
complete_task_state_url = f"{route.with_scheme(request.url.scheme).with_host(ip_addr).with_port(port)}"
```

Reply:
I tried both of these options but I encountered issues with both of them:
- when using the `route` directly I cannot call the `with_host`, `with_port` or `with_scheme` method because the route is relative. 
- When using the `AnyUrl.build` method I encounter a strange issue with the url encoding. When the value of the `future_id` parameter arrives in the endpoint which checks the state of the long running tasks which performs the multipart upload completion it is not url encoded correctly. It is not clear to me exactly what I would have to do when calling `AnyUrl.build` in order to avoid this.
The call which I already added works, so I prefer to keep it like this. In order to avoid the direct call to `AnyUrl` I use `parse_obj_as` which is the method which was also there before. Let me know if you don't like that.